jsonpatch.py

```diff
@@ -553,11 +553,16 @@ def __init__(self, patch, pointer_cls=JsonPointer):
         self.patch = patch
         self.pointer_cls = pointer_cls
 
+
         # Verify that the structure of the patch document
         # is correct by retrieving each patch element.
         # Much of the validation is done in the initializer
         # though some is delayed until the patch is applied.
         for op in self.patch:
+            if isinstance(op, basestring):
+                raise InvalidJsonPatch("Document is expected to be sequence of "
+                                       "operations, got a sequence of strings.")
+
```

Comment:
It seems overly specific to check for a basestring here. A number of types could be wrong here, not just strings. Wouldn't it make more sense to check whether it is *not* a list (or more generally a [``collections.abc.Sequence``](https://docs.python.org/3/library/collections.abc.html#collections.abc.Sequence))?

Reply:
That's what I though when writing the code. But two things kept me from doing it:

- If i's a JSON document we're iterating (which should be the case), we can only find strings as keys.
- I've spotted python-json-patch supports Python 2.7 and 3.5 and I'm not comfortable playing with ABCs on such old Pythons.

Reply:
OK, makes sense. Could you please add your first point (about strings being the only allowed keys) as a comment? I think this should help clarify things for anybody reading the code without the context of this pull request.
Thanks!

Reply:
Good idea! Hope it's not too verbose.